src/Http/Routing/src/Builder/RouteHandlerServices.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.Reflection;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing.Patterns;
+
+namespace Microsoft.AspNetCore.Routing;
+
+/// <summary>
+/// Provides methods used for invoking the route endpoint
+/// infrastructure with custom funcs for populating metadata
+/// and creating request delegates. Intended to be consumed from
+/// the RequestDeleatgeGenerator only.
+/// </summary>
+[EditorBrowsable(EditorBrowsableState.Never)]
+public static class RouteHandlerServices
```

Comment:
Can't this be added as a `file` class in the generated code, instead of being public that is not intended for developers to call?

Reply:
No, there's logic that we want to keep in the framework.